.github/workflows/style.yaml

```diff
@@ -37,6 +31,26 @@ jobs:
       run: |
         ruff check --output-format=github --no-fix .
         ruff format --diff
+  mypy:
+    name: mypy
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone repo
+        uses: actions/checkout@v4
+      - name: Set up python
+        id: setup-python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+      - name: Install pip dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          pip install -r requirements/requirements.txt -r requirements/style.txt -r requirements/tests.txt
+          pip cache purge
+      - name: List pip dependencies
+        run: pip list
+      - name: Run mypy checks
+        run: mypy --exclude lightning_uq_box/uq_methods .
```

Comment:
Technically, we don't have to exclude it both in `pyproject.toml` and in `style.yaml`, we could do one or the other. Which one we use depends on what you want the behavior of `mypy` locally to be. 


Next time I'm stuck on a train/plane/bus for several hours I'll try to tackle the uq_methods too, but not sure how soon that will be. Could also have a hackathon at the next happy hour if you just want to finish this together, I don't imagine it would take that long 🍻

Reply:
To me it is more clear that way of what the current status of mypy checks is. And yes, if we get an hour together somewhere, we could probably finish it.

Reply:
I usually try to keep all the config in the `pyproject.toml` so that the behavior is the same locally and in CI, but it's up to you.